test/EntityFramework6.Npgsql.Tests/Support/EntityFrameworkTestBase.cs

```diff
@@ -213,6 +223,85 @@ private static DbCompiledModel CreateModel(NpgsqlConnection connection)
+                    clone.SetMetadataProperties(x
+                        .MetadataProperties
+                        .Where(metadataProerty => !clone
+                            .MetadataProperties
+                            .Any(cloneMetadataProperty => cloneMetadataProperty.Name.Equals(metadataProerty.Name))));
+                    return clone;
+                }),
+                null);
+
+            var getBlogsFunc = EdmFunction.Create(
+                "StoredGetBlogsFunction",
+                "BloggingContext",
+                DataSpace.SSpace,
+                new EdmFunctionPayload
+                {
+                    ParameterTypeSemantics = ParameterTypeSemantics.AllowImplicitConversion,
+                    Schema = "dbo",
+                    IsComposable = true,
+                    IsNiladic = false,
+                    IsBuiltIn = false,
+                    IsAggregate = false,
+                    StoreFunctionName = "GetBlogsByName",
+                    ReturnParameters = new[]
+                    {
+                            FunctionParameter.Create("ReturnType1", rowType.GetCollectionType(), ParameterMode.ReturnValue)
+                    },
+                    Parameters = new[]
+                    {
+                            FunctionParameter.Create("Name", dbModel.ProviderManifest.GetStoreTypes().First(x=>x.ClrEquivalentType == typeof(String)), ParameterMode.In)
+                    }
+                },
+                null);
+            dbModel.StoreModel.AddItem(getBlogsFunc);
+
+            var modelBlogConceptualType = dbModel.ConceptualModel.EntityTypes.First(x => x.Name == typeof(Blog).Name) as EntityType;
+            EdmFunction getBlogsFuncModel = EdmFunction.Create(
+                "GetBlogsByName",
+                dbModel.ConceptualModel.Container.Name,
+                DataSpace.CSpace,
+                new EdmFunctionPayload
+                {
+                    IsFunctionImport = true,
+                    IsComposable = true,
+                    Parameters = new[] 
+                    {
+                        FunctionParameter.Create("Name", PrimitiveType.GetEdmPrimitiveTypes().First(x=>x.ClrEquivalentType == typeof(string)), ParameterMode.In)
+                    },
+                    ReturnParameters = new[]
+                    {
+                        FunctionParameter.Create("ReturnType1", modelBlogConceptualType.GetCollectionType(), ParameterMode.ReturnValue)
+                    },
+                    EntitySets = new[] 
+                    {
+                        dbModel.ConceptualModel.Container.EntitySets.First(x=>x.ElementType == modelBlogConceptualType)
+                    }
+                },
+                null);
+            dbModel.ConceptualModel.Container.AddFunctionImport(getBlogsFuncModel);
+
+            dbModel.ConceptualToStoreMapping.AddFunctionImportMapping(new System.Data.Entity.Core.Mapping.FunctionImportMappingComposable(
```

Comment:
Line length is a little too long, can you use a "using" and some formatting and make this a bit more readable ?